plugins/check_curl.c

```diff
@@ -324,9 +325,18 @@ int verify_callback(int preverify_ok, X509_STORE_CTX *x509_ctx)
 
 CURLcode sslctxfun(CURL *curl, SSL_CTX *sslctx, void *parm)
 {
-	(void) curl; // ignore unused parameter
-	(void) parm; // ignore unused parameter
-  SSL_CTX_set_verify(sslctx, SSL_VERIFY_PEER, verify_callback);
+  (void) curl; // ignore unused parameter
+  (void) parm; // ignore unused parameter
+  if(add_sslctx_verify_fun) {
+    SSL_CTX_set_verify(sslctx, SSL_VERIFY_PEER, verify_callback);
+  }
+
+  // workaround for issue:
+  // OpenSSL SSL_read: error:0A000126:SSL routines::unexpected eof while reading, errno 0
+  // see discussion https://github.com/openssl/openssl/discussions/22690
+#ifdef SSL_OP_IGNORE_UNEXPECTED_EOF
```

Comment:
Should that error be ignored unconditionaly?

Reply:
Things look fine to me, the only question I would be, how would you decide to set SSL_OP_IGNORE_UNEXPECTED_EOF at runtime? Always? Or only if the webserver is of a specific type?

Reply:
i'd just always ignore it. Doing this only for specific servers is probably way to much maintenance and adding a extra switch for it seems to as bit over-engineered.
And with that option in place, it behaves like check_http does as well.